### ISSUE TITLE
Could NOT find PerlModules (missing: PerlModules_all_modules_found   URI::Escape)

### ISSUE THREAD

**Issue by zkSNARK** [KDE-mac/homebrew-kde#182]
Could NOT find PerlModules (missing: PerlModules_all_modules_found
  URI::Escape)

repeatedly causing error in install for mac high sierra 10.13.1

**Comment by yurikoles** [comment]
https://github.com/KDE-mac/homebrew-kde/blob/24561d48151c923ac35b7c4375840e9a1d386916/kf5-kdelibs4support.rb#L45-L47

**Comment by jdknox** [comment]
After running the above commands I still get the `Could NOT find PerlModules_URI::Escape` error.  I also tried running the commands from [here](https://github.com/KDE-mac/homebrew-kde/pull/150/files) (but with `$(brew --prefix)` instead of `#{HOMEBREW_PREFIX}`.
In addition, I tried
```
# perl -MCPAN -e shell
cpan[1]> install URI::Escape
```
I removed the `mac-kde` tap, updated homebrew, retapped and tried reinstalling, and rebooted my machine to no avail.

I'm running macOS 10.13.2, perl 5.26.1

---

```
Last 15 lines from /Users/<USER>/Library/Logs/Homebrew/kf5-kdoctools/01.cmake:
-- Found Perl: /usr/local/bin/perl (found version "5.26.1") 
-- Could NOT find PerlModules_URI::Escape (missing: PerlModules_URI_Escape_FOUND) 
CMake Error at /usr/local/Cellar/cmake/3.10.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PerlModules (missing: PerlModules_all_modules_found
  URI::Escape)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.10.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPerlModules.cmake:77 (find_package_handle_standard_args)
  cmake/uriencode.cmake:34 (find_package)
  src/CMakeLists.txt:4 (kdoctools_encode_uri)


-- Configuring incomplete, errors occurred!
See also "/tmp/kf5-kdoctools-20180116-30407-1d92fcw/kdoctools-5.42.0/build/CMakeFiles/CMakeOutput.log".
See also "/tmp/kf5-kdoctools-20180116-30407-1d92fcw/kdoctools-5.42.0/build/CMakeFiles/CMakeError.log".

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/KDE-mac/homebrew-kde/issues
```

**Comment by vaboro** [comment]
It might be that cpanminus confuses perl about where to search for the URI::Escape module.
You may consider uninstalling cpanminus and [reinstalling](https://github.com/KDE-mac/homebrew-kde/issues/193) URI::Escape with cpan.

**Comment by gandarez** [comment]
Having the same problem when isntalling `kde-mac/kde/kf5-ktexteditor`

```
==> cmake -G Ninja .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/
Last 15 lines from /Users/gandarez/Library/Logs/Homebrew/kf5-kdoctools/01.cmake:
-- Found Perl: /usr/local/bin/perl (found version "5.30.0")
-- Could NOT find PerlModules_URI::Escape (missing: PerlModules_URI_Escape_FOUND)
CMake Error at /usr/local/Cellar/cmake/3.15.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PerlModules (missing: PerlModules_all_modules_found
  URI::Escape)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.15.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPerlModules.cmake:77 (find_package_handle_standard_args)
  cmake/uriencode.cmake:34 (find_package)
  src/CMakeLists.txt:4 (kdoctools_encode_uri)
```

**Comment by yurikoles** [comment]
Why don't you follow readme and doesn't use search?

**Comment by gandarez** [comment]
because didn't solve at all.. I ran these step and got another error

```
brew install cpanminus
cpanm URI
brew install kde-mac/kde/kf5-ktexteditor
```

```
==> Installing kde-mac/kde/kf5-ktexteditor
==> Downloading https://download.kde.org/stable/frameworks/5.61/ktexteditor-5.61.0.tar.xz
==> Downloading from http://kde.c3sl.ufpr.br/stable/frameworks/5.61/ktexteditor-5.61.0.tar.xz
######################################################################## 100.0%
==> cmake -G Ninja .. -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/
==> ninja
Last 15 lines from /Users/gandarez/Library/Logs/Homebrew/kf5-ktexteditor/02.ninja:
         <KTextEditor/document.h>
In file included from ../src/document/katedocument.cpp:59:
src/include/KTextEditor/Attribute:1:10: warning: non-portable path to file '"KTextEditor/attribute.h"'; specified path differs in case from file name on disk [-Wnonportable-include-path]
#include "ktexteditor/attribute.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~
         "KTextEditor/attribute.h"
74 warnings generated.
[193/199] Linking CXX shared library bin/libKF5TextEditor.5.61.0.dylib
[194/199] Creating library symlink bin/libKF5TextEditor.5.dylib bin/libKF5TextEditor.dylib
[195/199] Generating katepart.json
FAILED: src/part/katepart.json
cd /tmp/kf5-ktexteditor-20190829-35399-1fp7tcy/ktexteditor-5.61.0/src/part && /usr/local/bin/desktoptojson -i katepart.desktop -o /tmp/kf5-ktexteditor-20190829-35399-1fp7tcy/ktexteditor-5.61.0/build/src/part/katepart.json -s kpart.desktop
Warning: Could not locate service type file kservicetypes5/kpart.desktop, tried ("/Users/gandarez/Library/Application Support", "/Library/Application Support") and ":/kservicetypes5/kpart.desktop" ((null):0, (null))
About to parse service type file "kpart.desktop"
ninja: build stopped: subcommand failed.
```

**Comment by yurikoles** [comment]
Readme states that you need to execute


```sh
"$(brew --repo kde-mac/kde)/tools/do-caveats.sh"
```

**Before** installing anything. Is it that hard to read one page?

**Comment by rgudwin** [comment]
If we look at the /usr/local/Homebrew/Library/Taps/kde-mac/homebrew-kde/tools/do-caveats.sh script, we will see that its first lines are the following:

brew install perl dbus
export PERL_MM_USE_DEFAULT=1
cpan CPAN
cpan URI URI::Escape

So, what this script is doing is installing perl (and dbus) and then installing these two perl modules: URI and URI::Escape. 
In my case, I started with a completely fresh HomeBrew installation, and the very first movements where to do:
brew tap kde-mac/kde
$(brew --repo kde-mac/kde)/tools/do-caveats.sh
as is instructed at: https://github.com/KDE-mac/homebrew-kde
Nevertheless, during installation of perl, I got a problem and had to do a CTRL-C in the middle of the installation and start it again. After running for the 2nd time, it concluded without errors, but apparently this created a problem in my perl installation, and put myself in a strange condition, which I was not able to recover. Even though I installed manually the URI::Escape module using:
cpan URI URI::Escape, for some reason, it was installed in a directory which is different from the one searched for during the brew install <KDE_PACKAGE>. The way I found to escape from this extraneous condition was to do:
/usr/local/Cellar/perl/5.30.0/bin
and then:
./cpan install URI::Escape
Apparently, doing so, I was able to recover with the original installation of perl and install the URI and URI::Escape in the right directory for the brew install <KDE_APP> to work. 
As I could see that many users were reporting this same problem, apparently this is not something which happened just to me, but apparently a corruption in the perl installation could lead to this problem ... and this is a possible solution to solve the problem. 
I hope this could be useful to others.
